packages/next/next-server/lib/router/router.ts

```diff
@@ -491,18 +493,30 @@ export default class Router implements BaseRouter {
     if (!(options as any)._h && this.onlyAHashChange(cleanedAs)) {
       this.asPath = cleanedAs
       Router.events.emit('hashChangeStart', as)
+      // TODO: do we need the resolved href when only a hash change?
       this.changeState(method, url, as, options)
       this.scrollToHash(cleanedAs)
       this.notify(this.components[this.route])
       Router.events.emit('hashChangeComplete', as)
       return true
     }
 
-    const parsed = tryParseRelativeUrl(url)
+    const pages = await this.pageLoader.getPageList()
```

Comment:
What happens if this fails?

Reply:
Currently if this fails it will reject, do you think we should handle this failing the same way we handle the page bundle failing to load?

Reply:
Reloading from the server would just re-trigger this error in an infinite loop AFAIK. We probably have to render _error.